### PR TITLE
net/redistest: run redis containers on test network

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Containers on test network will use the same port and different addresses unlike containers on the host network, which use different ports and the same loopback address.

### CODE REVIEW

File: net/redistest/redistest.go [hunk 1]
```diff
@@ -2,15 +2,29 @@ package redistest
...
+// testNetwork returns a network for test containers.
+// Containers on this network will use the same port and different addresses
+// unlike containers on the host network, which use different ports and the same loopback address.
+// The network will be cleaned up automatically by testcontainers at exit.
+var testNetwork = sync.OnceValues(func() (*testcontainers.DockerNetwork, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	return network.New(ctx)
+})
```
Comment (AlexanderYastrebov) [inline net/redistest/redistest.go:26]: This does not pass noleak :(
```
noleak: 1 active

goroutine 12800 [chan receive]:
github.com/testcontainers/testcontainers-go.(*Reaper).Connect.func1({0x1b4f8f8, 0xc000526650})
	/home/runner/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.27.0/reaper.go:360 +0x76e
created by github.com/testcontainers/testcontainers-go.(*Reaper).Connect in goroutine 12769
	/home/runner/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.27.0/reaper.go:325 +0x170
```
Need to call network.Remove() after last container stops using the network.

Comment (AlexanderYastrebov) [reply net/redistest/redistest.go]: Fixed

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**